hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -395,9 +418,13 @@ private void updateKeys(
             validateTrue(originalToken.hasMetadataKey(), TOKEN_HAS_NO_METADATA_KEY);
             builder.metadataKey(op.metadataKey());
         }
-        if (!isExpiryOnlyUpdateOp(op)) {
-            validateTrue(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
+
+        if (isMetadataOnlyUpdateOp(op)) {
+            validateTrue(originalToken.hasAdminKey() || originalToken.hasMetadataKey(), TOKEN_IS_IMMUTABLE);
+        } else if (!isExpiryOnlyUpdateOp(op)) {
+            validateFalse(originalToken.hasAdminKey(), TOKEN_IS_IMMUTABLE);
```

Comment:
Shouldn't this be `validateFalse(isEmpty(token.adminKey()), TOKEN_IS_IMMUTABLE);`?

Are these checks necessary at all? We have already done them in `validateSemantics()`.